ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,123 @@ const base = url.substring(0, url.lastIndexOf('/'));
 
-  test('layout stamped with model', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
-    `);
-    await waitChanged(layout, 'element');
-    expect(layout.$$('nuxeo-error').hidden).to.be.true;
-    expect(layout.element.localName).to.equal('dummy-layout');
-    expect(layout.element.text).to.equal('dummy');
-  });
+suite('<nuxeo-layout>', () => {
+  suite('Error handling', () => {
+    test('Should display nuxeo-error when layout is not found', async () => {
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href="notfound.html"></nuxeo-layout>
+        `,
+      );
+      await flush();
 
-  test('layout valid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
-  });
+      const nuxeoError = layout.$.error;
+      if (!isElementVisible(nuxeoError)) {
+        await waitForAttrMutation(nuxeoError, 'hidden', null);
+      }
+      expect(isElementVisible(layout.$.container)).to.be.false;
+      expect(isElementVisible(nuxeoError)).to.be.true;
+    });
 
-  test('layout valid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(true);
+    test('Should hide layout when href is empty', async () => {
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href=""></nuxeo-layout>
+        `,
+      );
+      await flush();
+
+      expect(isElementVisible(layout)).to.be.false;
+    });
   });
 
-  test('layout invalid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Stamping', () => {
+    test('Should stamp the layout when model is defined', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      expect(isElementVisible(layout.$.error)).to.be.false;
+      expect(layout.element.tagName).to.equal('DUMMY-LAYOUT');
+      expect(layout.element.shadowRoot.querySelector('span')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('span').innerText).to.equal('dummy');
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.exist;
+    });
+
+    test('Should stamp the layout when model is empty', async () => {
```

Comment:
Just look at tests description, can I assume it should stamp anyway? Despite the model being there, or not.
What differs is the mapping, correct?

Reply:
That is correct. Both should stamp, but in the first test we are making sure our model is being properly set.

Reply:
So, we may reword the description since it should be self explanatory.
`Should map model into layout when model is not empty`?